### ISSUE TITLE
Docker requiered?

### ISSUE THREAD

**Issue by CodeInFilth** [spartan737/Stocksera#12]
```
    output = _run(java_options, kwargs, path, encoding)
  File "/usr/local/lib/python3.7/dist-packages/tabula/io.py", line 91, in _run
    raise JavaNotFoundError(JAVA_NOT_FOUND_ERROR)
tabula.errors.JavaNotFoundError: `java` command is not found from this Python process.Please ensure Java is installed and PATH is set for `java`
```

is docker now a requirement to run?

**Comment by spartan737** [comment]
No, docker is never a requirement to run. I didn’t make any changes. Which lines are causing this error?

**Comment by CodeInFilth** [comment]
```bash
Traceback (most recent call last):
  File "tasks_to_run.py", line 142, in <module>
    get_upcoming_events_date.main()
  File "/root/_production_website/Stocksera/scheduled_tasks/economy/get_upcoming_events_date.py", line 66, in main
    retail_df = get_next_retail_sales_date()
  File "/root/_production_website/Stocksera/scheduled_tasks/economy/get_upcoming_events_date.py", line 17, in get_next_retail_sales_date
    df = tabula.read_pdf(r"https://www.census.gov/retail/marts/www/martsdates.pdf", pages=1)[0]
  File "/usr/local/lib/python3.7/dist-packages/tabula/io.py", line 322, in read_pdf
    output = _run(java_options, kwargs, path, encoding)
  File "/usr/local/lib/python3.7/dist-packages/tabula/io.py", line 91, in _run
    raise JavaNotFoundError(JAVA_NOT_FOUND_ERROR)
tabula.errors.JavaNotFoundError: `java` command is not found from this Python process.Please ensure Java is installed and PATH is set for `java`
```

Seems like I can hide lines and information until it works but this Java runtime error happens in multiple places when I work around the software.


Here is a full error, ( I ran again not sure what differs from what I submitted in the first comment. I am going to backup, clone and try again but unless I posted here later I am stuck still

**Comment by CodeInFilth** [comment]
> No, docker is never a requirement to run. I didn’t make any changes.
> 
> It seems like you need Java to run scheduled_tasks/economy/get_upcoming_events_date.py
> 
> refer to [stackoverflow.com/questions/56146587/how-to-fix-this-error-on-tabula-read-pdf-function-in-python#comment98923161_56146587](https://stackoverflow.com/questions/56146587/how-to-fix-this-error-on-tabula-read-pdf-function-in-python#comment98923161_56146587) for more info.

So here I removed and backed up my old one I updated and did a fresh install and the errors still here lines changed slightly.

```zsh
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/tabula/io.py", line 85, in _run
    check=True,
  File "/usr/lib/python3.7/subprocess.py", line 488, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'java': 'java'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "_daily_tasks_to_run.py", line 141, in <module>
    get_upcoming_events_date.main()
  File "/root/_production_website/Stocksera/scheduled_tasks/economy/get_upcoming_events_date.py", line 66, in main
    retail_df = get_next_retail_sales_date()
  File "/root/_production_website/Stocksera/scheduled_tasks/economy/get_upcoming_events_date.py", line 17, in get_next_retail_sales_date
    df = tabula.read_pdf(r"https://www.census.gov/retail/marts/www/martsdates.pdf", pages=1)[0]
  File "/usr/local/lib/python3.7/dist-packages/tabula/io.py", line 322, in read_pdf
    output = _run(java_options, kwargs, path, encoding)
  File "/usr/local/lib/python3.7/dist-packages/tabula/io.py", line 91, in _run
    raise JavaNotFoundError(JAVA_NOT_FOUND_ERROR)
tabula.errors.JavaNotFoundError: `java` command is not found from this Python process.Please ensure Java is installed and PATH is set for `java`
```

Here is an text upload of my full console output:
	https://0bin.net/paste/nnjbKLbV#Jk7Lx21JURRiEmmb5g3VqjxO34oY+so0lj-1wh4X6Q+
	
Here is the tasks_to_run.py with the small changes I have made:
	https://0bin.net/paste/Jw0V2EUY#J74m1pOkaP3fOlg0gHsFK2IkWcVH+W-tqpH+tvhSzZd
	
	
P.S. -- I am also using unix OS, I have been taking notes on the nenssicary changes and differnces for this to run on Ubuntu 18.04+ and Mac OS Big Sur 11.4+ (What I have tested on)

**Comment by CodeInFilth** [comment]
Solved!

Idk what made me think this was a docker error now lmao

Ubuntu Fix (tesed on 18.04 and 20.04)
	```bash
	➜ $ sudo apt install default-jdk
	```
Mac OS Fix:
	requires `brew` and `cask` to be installed for ease of use for users
	  
	```bash
	➜ $ brew install --cask adoptopenjdk/openjdk/adoptopenjdk11
	```
